eos-updater/apply.c

```diff
@@ -227,6 +228,65 @@ update_remote_branches (OstreeRepo   *repo,
   return TRUE;
 }
 
+static gboolean
+sysroot_boot_is_automount (OstreeSysroot *sysroot)
+{
+  GFile *sysroot_file;
+  g_autoptr(GFile) boot_file = NULL;
+  g_autofree gchar *boot_path = NULL;
+  gboolean ret = FALSE;
+  struct libmnt_table *tb = NULL;
+  struct libmnt_cache *cache;
+  struct libmnt_fs *fs;
+
+  sysroot_file = ostree_sysroot_get_path (sysroot);
+  boot_file = g_file_get_child (sysroot_file, "boot");
+  boot_path = g_file_get_path (boot_file);
+  if (boot_path == NULL)
+    {
+      g_autofree gchar *sysroot_path = g_file_get_path (sysroot_file);
+      g_warning ("No boot directory in sysroot %s", sysroot_path);
+      goto out;
+    }
+
+  tb = mnt_new_table_from_file ("/proc/self/mountinfo");
+  if (tb == NULL)
+    {
+      g_warning ("Failed to parse /proc/self/mountinfo: %s",
+                 g_strerror (errno));
+      goto out;
+    }
+
+  cache = mnt_new_cache ();
+  if (cache == NULL)
+    {
+      g_warning ("Failed to create libmount cache: %s", g_strerror (errno));
+      goto out;
+    }
+  mnt_table_set_cache (tb, cache);
+  mnt_unref_cache (cache);
+
+  /* Find the /boot mountpoint iterating forward so that the autofs mount is
+   * found and not the automount target filesystem.
```

Comment:
It took me a little bit to understand how this works -- perhaps you could explain it a bit more in the comment? Maybe something like:
```suggestion
  /* Find the /boot mountpoint iterating forward so that the autofs mount is
   * found and not the automount target filesystem, as the autofs mount is
   * listed first in mountinfo.
```

Reply:
:+1: